subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+{
+	switch (hci_opcode) {
+#if defined(CONFIG_BT_BROADCASTER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_ADV_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_ADV_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_RESPONSE_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_ADV_ENABLE:
+	case SDC_HCI_OPCODE_CMD_LE_READ_MAX_ADV_DATA_LENGTH:
+	case SDC_HCI_OPCODE_CMD_LE_READ_NUMBER_OF_SUPPORTED_ADV_SETS:
+	case SDC_HCI_OPCODE_CMD_LE_REMOVE_ADV_SET:
+	case SDC_HCI_OPCODE_CMD_LE_CLEAR_ADV_SETS:
+#endif  /* CONFIG_BT_BROADCASTER */
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_ENABLE:
+#endif  /* CONFIG_BT_OBSERVER */
+#if defined(CONFIG_BT_CENTRAL)
+	case SDC_HCI_OPCODE_CMD_LE_EXT_CREATE_CONN:
+#endif
+		if (type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_NONE) {
+			type_of_adv_cmd_used_since_reset = ADV_COMMAND_TYPE_EXTENDED;
+			return false;
+		}
+		return type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_LEGACY;
+
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_ADV_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_READ_ADV_PHYSICAL_CHANNEL_TX_POWER:
+	case SDC_HCI_OPCODE_CMD_LE_SET_ADV_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_SCAN_RESPONSE_DATA:
+	case SDC_HCI_OPCODE_CMD_LE_SET_ADV_ENABLE:
+#endif  /* CONFIG_BT_OBSERVER */
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_SCAN_PARAMS:
+	case SDC_HCI_OPCODE_CMD_LE_SET_SCAN_ENABLE:
+#endif  /* CONFIG_BT_OBSERVER */
+#if defined(CONFIG_BT_CENTRAL)
+	case SDC_HCI_OPCODE_CMD_LE_CREATE_CONN:
+#endif  /* CONFIG_BT_CENTRAL */
+		if (type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_NONE) {
+			type_of_adv_cmd_used_since_reset = ADV_COMMAND_TYPE_LEGACY;
+			return false;
+		}
+		return type_of_adv_cmd_used_since_reset == ADV_COMMAND_TYPE_EXTENDED;
+	case SDC_HCI_OPCODE_CMD_CB_RESET:
+		type_of_adv_cmd_used_since_reset = ADV_COMMAND_TYPE_NONE;
+		break;
+	default:
+		/* Ignore command */
+		break;
+	}
+
+	return false;
+}
+
+static void encode_command_status(uint8_t * const event,
+				  uint16_t hci_opcode,
+				  uint8_t status_code)
+{
+
+	struct bt_hci_evt_hdr *evt_hdr = (void *)event;
+	struct bt_hci_evt_cmd_status *evt_data = (void *)&event[BT_HCI_EVT_HDR_SIZE];
+
+	evt_hdr->evt = BT_HCI_EVT_CMD_STATUS;
+	evt_hdr->len = BT_HCI_EVT_HDR_SIZE + sizeof(struct bt_hci_evt_cmd_status);
+
+	evt_data->status = status_code;
+	evt_data->ncmd = 1;
+	evt_data->opcode = hci_opcode;
+}
+
+static void encode_command_complete_header(uint8_t * const event,
+					   uint16_t hci_opcode,
+					   uint8_t param_length,
+					   uint8_t status)
+{
+	struct bt_hci_evt_hdr *evt_hdr = (void *)event;
+	struct bt_hci_evt_cmd_complete *evt_data = (void *)&event[BT_HCI_EVT_HDR_SIZE];
+
+	evt_hdr->evt = BT_HCI_EVT_CMD_COMPLETE;
+	evt_hdr->len = param_length;
+	evt_data->ncmd = 1;
+	evt_data->opcode = hci_opcode;
+	event[BT_HCI_EVT_HDR_SIZE + sizeof(struct bt_hci_evt_cmd_complete)] = status;
```

Comment:
Might be unrelated, but why is there no type that includes status field as well?

Reply:
There a some in https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/hci.h, however, those are specialized for the given commands